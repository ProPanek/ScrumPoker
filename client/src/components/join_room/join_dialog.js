import React, {Component} from "react"
import Dialog from "@material-ui/core/Dialog"
import DialogTitle from "@material-ui/core/DialogTitle"
import DialogContent from "@material-ui/core/DialogContent"
import DialogContentText from "@material-ui/core/DialogContentText"
import TextField from "@material-ui/core/TextField"
import DialogActions from "@material-ui/core/DialogActions"
import Button from "@material-ui/core/Button"
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import {decorate, observable} from "mobx";
import routes from "routes"

class JoinDialog extends Component {
  constructor(props) {
    super(props)
    this.userName = ""
    this.roomPassword = ""
  }

  componentDidMount() {
    // if (localStorage.getItem("userName") !== null) {
    //   this.userName = JSON.parse(localStorage.getItem("userName"))
    // }
  }

  handleChangUserName = e => {
    this.userName = e.target.value
  };

  handleChangRoomPassword = e => {
    this.roomPassword = e.target.value
  };
  joinRoom = () => {
    const {match: {params}, store: {roomStore: {joinRoom}, socketStore: {openNotification}, userStore: {setUserName}}} = this.props
    if (this.userName && this.roomPassword) {
      setUserName(this.userName)
      joinRoom(this.userName, params.id, this.roomPassword)
    } else {
      openNotification("To join a room you need to fill all inputs", "warning")
    }
  }

  cancelJoinRoom = () => {
    this.props.history.push(routes.root())
  }

  render() {
    const {store: {userStore: {isConnected}}} = this.props
    return (
      <Dialog
        open={!isConnected}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Join Room</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Please write your username below if you want to join to the room.
          </DialogContentText>
          <TextField
            autoFocus
            margin="dense"
            id="userName"
            label="User Name"
            type="text"
            fullWidth
            value={this.userName}
            onChange={this.handleChangUserName}
          />
          <TextField
            autoFocus
            margin="dense"
            id="roomPassword"
            label="Room Password"
            type="password"
            fullWidth
            value={this.roomPassword}
            onChange={this.handleChangRoomPassword}
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={this.cancelJoinRoom} color="primary">
            Cancel Join
          </Button>
          <Button onClick={this.joinRoom} color="primary" variant="contained">
            Join Room
          </Button>
        </DialogActions>
      </Dialog>
    )
  }
}

decorate(JoinDialog, {
  userName: observable,
  roomPassword: observable
});

export {JoinDialog}
export default inject("store")(withRouter(observer(JoinDialog)));

