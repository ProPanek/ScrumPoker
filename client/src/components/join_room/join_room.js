import React, {Component} from "react";
import {Grid, Card, Button, TextField} from "@material-ui/core";
import styled from "styled-components";
import {inject, observer} from "mobx-react";
import DeleteIcon from "@material-ui/icons/Delete"
import {ArrowUpBold} from "mdi-material-ui"
import Typography from "@material-ui/core/Typography/Typography";
import {decorate, observable, reaction} from "mobx";
import routes from "routes"

const StyledGrid = styled(Grid)`
  height: calc(100vh - 64px);
`;

const StyledCard = styled(Card)`
  height: calc(100vh - 64px);
`;

const Wrapper = styled.div`
  @media only screen and (max-width: 768px) {
     width: 90%;
  }
  display: flex;
  flex-direction: column;
  width: 50%;
  margin: 0 auto;
  position: relative;
  top: calc(50vh - 240px);
`;

const StyledButton = styled(Button)`
  &&{
  width:100%;
  }
`;

class JoinRoom extends Component {
  constructor(props) {
    super(props)
    this.userName = ""
    this.roomId = ""
    this.roomPassword = ""
  }

  componentDidMount() {
    // if (localStorage.getItem("userName") !== null) {
    //   this.setState({userName: JSON.parse(localStorage.getItem("userName"))});
    // }
    reaction(
      () => this.props.store.userStore.connected,
      () => {
        if (this.props.store.userStore.connected) {
          this.props.history.push(routes.room(this.roomName, this.roomId))
        }
      },
    )
  }

  handleChangeUserName = e => {
    this.userName = e.target.value
  };

  handleChangeRoomId = e => {
    this.roomId = e.target.value
  };

  handleChangeRoomPassword = e => {
    this.roomPassword = e.target.value
  };

  handleSubmit = () => {
    const {store: {roomStore: {joinRoom}, socketRoom: {openNotification}}} = this.props
    if (this.roomPassword && this.roomId && this.userName) {
      joinRoom(this.userName, this.roomId, this.roomPassword);
    } else {
      openNotification("To join a room you need to fill all inputs", "warning")
    }
  };

  handleDelete = () => {
    const {store: {roomStore: {deleteRoom}}} = this.props
    if (this.roomPassword && this.roomId) {
      deleteRoom(
        this.roomId,
        this.roomPassword
      );
    }
  };

  render() {
    return (
      <StyledGrid item xs={12}>
        <StyledCard>
          <Wrapper>
            <Typography variant="h3" align={"center"}>
              Join Room
            </Typography>
            <TextField
              id="join-user-name"
              label="User Name"
              value={this.userName}
              onChange={this.handleChangeUserName}
              margin="normal"
            />
            <TextField
              id="join-room-id"
              label="Room id"
              placeholder="ex. 11a2bae1-1b9b-4807-9db6-c54c51989fe9"
              value={this.roomId}
              onChange={this.handleChangeRoomId}
              margin="normal"
            />
            <TextField
              id="join-room-password"
              label="Room Password"
              value={this.roomPassword}
              onChange={this.handleChangeRoomPassword}
              type="password"
              margin="normal"
            />
            <Grid style={{marginTop: "40px"}} container>
              <Grid item xs={6}>
                <StyledButton color="primary" variant="contained" onClick={this.handleSubmit}>
                  Join room
                  <ArrowUpBold style={{marginLeft: "10px"}}/>
                </StyledButton>
              </Grid>
              <Grid item xs={6}>
                <StyledButton color="secondary" variant="contained" onClick={this.handleDelete}>
                  Delete room
                  <DeleteIcon style={{marginLeft: "10px"}}/>
                </StyledButton>
              </Grid>
            </Grid>
          </Wrapper>
        </StyledCard>
      </StyledGrid>
    );
  }
}

decorate(JoinRoom, {
  userName: observable,
  roomId: observable,
  roomPassword: observable
});

export {JoinRoom};
export default inject("store")(observer(JoinRoom));
