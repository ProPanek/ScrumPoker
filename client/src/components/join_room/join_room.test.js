import React from "react";
import {JoinRoom} from "./join_room";
import { shallow} from "enzyme";


const noop = () => {}
describe("JoinRoom", () => {

  it("renders without crashing", () => {
    const props = {
      store: {
        userStore: {connected: false}
      }
    }
    const component = shallow(<JoinRoom {...props}/>)
    expect(component).toMatchSnapshot()
  });

});