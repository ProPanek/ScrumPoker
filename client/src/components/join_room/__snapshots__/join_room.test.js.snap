// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinRoom renders without crashing 1`] = `
<Styled(WithStyles(Grid))
  item={true}
  xs={12}
>
  <Styled(WithStyles(Card))>
    <styled.div>
      <WithStyles(Typography)
        align="center"
        variant="h3"
      >
        Join Room
      </WithStyles(Typography)>
      <TextField
        id="join-user-name"
        label="User Name"
        margin="normal"
        onChange={[Function]}
        required={false}
        select={false}
        value=""
        variant="standard"
      />
      <TextField
        id="join-room-id"
        label="Room id"
        margin="normal"
        onChange={[Function]}
        placeholder="ex. 11a2bae1-1b9b-4807-9db6-c54c51989fe9"
        required={false}
        select={false}
        value=""
        variant="standard"
      />
      <TextField
        id="join-room-password"
        label="Room Password"
        margin="normal"
        onChange={[Function]}
        required={false}
        select={false}
        type="password"
        value=""
        variant="standard"
      />
      <WithStyles(Grid)
        container={true}
        style={
          Object {
            "marginTop": "40px",
          }
        }
      >
        <WithStyles(Grid)
          item={true}
          xs={6}
        >
          <Styled(WithStyles(Button))
            color="primary"
            onClick={[Function]}
            variant="contained"
          >
            Join room
            <Icon
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            />
          </Styled(WithStyles(Button))>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          xs={6}
        >
          <Styled(WithStyles(Button))
            color="secondary"
            onClick={[Function]}
            variant="contained"
          >
            Delete room
            <pure(DeleteIcon)
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            />
          </Styled(WithStyles(Button))>
        </WithStyles(Grid)>
      </WithStyles(Grid)>
    </styled.div>
  </Styled(WithStyles(Card))>
</Styled(WithStyles(Grid))>
`;
